inlineplz/linters/__init__.py

```diff
@@ -650,268 +645,285 @@ def vendored_path(path):
+        paths = self.all_filenames_in_dir(path=path)
         for pattern in patterns:
-            if fnmatch.filter(filenames, pattern):
-                return True
+            for filepath in fnmatch.filter(paths, pattern):
+                if "text" in identify.tags_from_path(filepath):
+                    run_cmds.append(cmd + [filepath])
+                    file_tasks.append(self.event_loop.create_task(self.run_command(run_cmds[-1], timeout=5)))
 
-    return False
+        output = []
+        for run_cmd, file_task in zip(run_cmds, file_tasks):
+            _, out = await file_task
 
+            # Return the filename and the output
+            output.append((run_cmd[-1], out.strip()))
 
-def dotfiles_exist(config, path=None):
-    path = path or os.getcwd()
-    return any(
-        dotfile.strip() in os.listdir(path) for dotfile in config.get("dotfiles")
-    )
+        return output
 
+    def linters_to_run(self):
+        linters = set()
+        enabled_linters = self.enabled_linters or []
+        disabled_linters = self.disabled_linters or []
+        try:
+            enabled_linters.extend(enabled_linters[0].split(","))
+        except (IndexError, AttributeError):
+            pass
+        try:
+            disabled_linters.extend(disabled_linters[0].split(","))
+        except (IndexError, AttributeError):
+            pass
+        if not self.autorun:
+            for linter, config in LINTERS.items():
+                if linter in enabled_linters:
+                    linters.add(linter)
+        else:
+            dotfilefound = {}
+            for linter, config in LINTERS.items():
+                if self.dotfiles_exist(config):
+                    dotfilefound[config.get("language")] = True
+                    if (
+                        config.get("run_if_dotfile_in_root")
+                        and linter not in disabled_linters
+                    ):
+                        linters.add(linter)
+                if dotfilefound.get(config.get("language")) and config.get("autorun"):
+                    if linter not in disabled_linters:
+                        linters.add(linter)
+            filenames = self.all_filenames_in_dir()
+            for linter, config in LINTERS.items():
+                if linter in enabled_linters or (
+                    not dotfilefound.get(config.get("language"))
+                    and self.should_autorun(config, filenames)
+                ):
+                    if linter not in disabled_linters:
+                        linters.add(linter)
+        return linters
+
+    def all_filenames_in_dir(self, path=None):
+        path = path or os.getcwd()
+        # http://stackoverflow.com/a/2186565
+        paths = set()
+        for root, dirnames, filenames in os.walk(path, topdown=True):
+            try:
+                for ignore in self.ignore_paths:
+                    dirnames.remove(ignore)
+            except ValueError:
```

Comment:
[inline-plz]: `prospector: mccabe: LinterRunner.linters_to_run is too complex (16) (MC0001)`